### ISSUE TITLE
One morphology of release can't be loaded in NEURON

### ISSUE THREAD

**Issue by wvangeit** [lbologna/bsp_data_repository#10]
We have one morphology in the hippocampus morph release that fails when loaded in NEURON with:
82000 lines read/Users/werner/local/nrngui/x86_64/bin/nrniv: eof call nested too deeply, increase with -NFRAME framesize option
near line 1
{readfile()}
            ^
       Import3d_Neurolucida3[0].getline("          ...")
     Import3d_Neurolucida3[0].lex(Import3d_LexToken[18])
   Import3d_Neurolucida3[0].read_next_token_lex()
 Import3d_Neurolucida3[0].read_next_token()
and others

/gpfs/bbp.cscs.ch/project/proj42/CircuitRelease/r2017.06/MorphologyRelease/rat/CA1/06_GraftAxons-asc/970428A1.asc

**Comment by lbologna** [comment]
Hi Werner,

here's Michele's answer:

"
morphologies giving this error most likely have a highly branching axonal tree which exceeds default NEURON values. Setting -NFRAME 1000 in the NEURON launch command line should avoid this (see last lines in the nrndef file in the NEURON distribution).
We did not consider any of these morphologies in our optimizations.
Michele Migliore
"

**Comment by wvangeit** [comment]
Yes, indeed the NFRAME has to be set, but this can not be done from within Python as far as I know. So to avoid future problems we should remove this morph from the hippocampus morphology release.

**Comment by romani79** [comment]
In the worst case, we can remove this morph since you have another exemplar of the same m-type. Anyway, is there no way to fix this issue without removing the morph?

**Comment by wvangeit** [comment]
well, we could use a modified version of NEURON. But I don't think that's a good idea, because this problem will percolate all along the workflow, and every step that ever loads this morph in NEURON would have to use the same modified version (I'm talking about e.g. VMs in the collab etc.)

One thing that 'might' work, is to merge some fragments in the morphology together (because I assume the error is caused by having too many lines / fragments in the file, not 100% sure though)

**Comment by liesbethvanherpe** [comment]
@romani79, do you have an update on the use of morphology 970428A1? Thanks.

**Comment by romani79** [comment]
> In the worst case, we can remove this morph since you have another exemplar of the same m-type. 

I have to correct myself: 970428A1 should not be removed. It is true I have two cells for OLM, but they are two different subtypes of OLM cells.

> One thing that 'might' work, is to merge some fragments in the morphology together

I do not know... this does not sound like an ideal solution to me and we are not even sure it is going to work.

**Comment by wvangeit** [comment]
> I do not know... this does not sound like an ideal solution to me and we are not even sure it is going to work.

But I'm afraid I don't see any other solution if you want to keep this morph ...

**Comment by mmigliore** [comment]
We can’t stop the entire pipeline for this, relatively minor, problem. I strongly suggest to remove it and proceed, please.

Thank you

Michele

 

 

Da: romani79 [mailto:notifications@github.com] 
Inviato: 26 February, 2018 9:30 AM
A: lbologna/bsp_data_repository <bsp_data_repository@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Oggetto: Re: [lbologna/bsp_data_repository] One morphology of release can't be loaded in NEURON (#10)

 

In the worst case, we can remove this morph since you have another exemplar of the same m-type.

I have to correct myself: 970428A1 should not be removed. It is true I have two cells for OLM, but they are two different subtypes of OLM cells.

One thing that 'might' work, is to merge some fragments in the morphology together

I do not know... this does not sound like an ideal solution to me and we are not even sure it is going to work.
